### PR TITLE
feat: add OpenAI MCP (Model Context Protocol) support to Responses API

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary

This PR adds comprehensive support for OpenAI's Model Context Protocol (MCP) integration to the Responses API provider, enabling direct connection to remote MCP servers.

### Core Implementation
- **OpenAI Responses Provider**: Added MCP tool support with proper request/response handling
- **Type Definitions**: Added `OpenAiMCPTool` interface for MCP tool configuration  
- **Response Processing**: Enhanced response handler to process MCP-specific output types (`mcp_list_tools`, `mcp_call`, `mcp_approval_request`)

### Examples
- **New Example Directory**: `examples/openai-mcp/` with comprehensive MCP integration examples
- **Basic MCP Usage**: Public DeepWiki server integration example
- **Authenticated MCP**: Stripe server integration with API key authentication
- **Approval Workflows**: Examples showing different approval settings and strategies
- **Updated OpenAI Responses**: Added MCP example to existing responses examples

### Documentation  
- **OpenAI Provider Docs**: Added comprehensive MCP section with configuration options, authentication, and examples
- **MCP Integration Docs**: Added cross-reference to OpenAI's native MCP support
- **README Updates**: Improved example structure and documentation following project guidelines

### Testing
- **Comprehensive Test Suite**: Added 8+ test cases covering MCP tool configuration, response handling, authentication, and edge cases
- **Assertion Fixes**: Updated OpenAI tools assertion to properly filter MCP tools from function tools

### MCP Tool Configuration
- Remote MCP server connections via `server_url`
- Authentication via custom `headers`
- Tool filtering with `allowed_tools`  
- Flexible approval settings (`never`, selective by tool name)

### Response Handling
- `mcp_list_tools`: Tool discovery and visibility
- `mcp_call`: Tool execution results and errors
- `mcp_approval_request`: User approval workflows

### Security & Approval
- Default approval required for data sharing
- Selective approval by tool name
- Secure API key handling via environment variables

## Examples Usage

```bash
# Initialize the example
npx promptfoo@latest init --example openai-mcp

# Run basic MCP integration
npx promptfoo eval -c promptfooconfig.yaml

# Run authenticated MCP servers
npx promptfoo eval -c promptfooconfig.authenticated.yaml

# Run approval workflow examples  
npx promptfoo eval -c promptfooconfig.approval.yaml
```

## Breaking Changes
None - this is a purely additive feature.

## Testing
- All existing tests pass
- New MCP-specific test suite with comprehensive coverage
- Examples tested with latest model versions (gpt-4.1-2025-04-14)

## Documentation
- Follows project documentation guidelines
- Updated model versions to 2025 standards
- Cross-references between general MCP and OpenAI-specific implementations

### CODE REVIEW

File: examples/openai-mcp/promptfooconfig.yaml
Comment (adelmuursepp) [inline examples/openai-mcp/promptfooconfig.yaml:1]: I think that for the intro simple Responses API test it should have less prompts and MCPs. Responses API has much lower limits that Completions and after a few runs of `promptfoo eval` I get rate limited by OpenAI. Maybe split different MCPs in different files.

Also it takes very long to compose answers because of multiturn and long answer -> for a simple sample limit tokens to 500.

Comment (adelmuursepp) [reply examples/openai-mcp/promptfooconfig.yaml:1]: Probably more relevant for users with personal and not org accounts. Just useful for better UX because right now promptfoo just says failed after 4 retries. It doesn't show error 429 from OpenAI.

File: test/assertions/openai.test.ts [hunk 1]
```diff
@@ -1045,4 +1045,287 @@ describe('OpenAI assertions', () => {
...
+      });
+
+      expect(result).toEqual({
+        pass: false,
+        score: 0,
+        reason: 'MCP tool call failed for ask_question: Repository not found',
+        assertion: toolsAssertion,
+      });
+    });
+
```
Comment (adelmuursepp) [inline test/assertions/openai.test.ts:1096]: Some tests like this seem a little unnecessary, especially since that is not a very complex tool name. But does not hurt I guess.
